### PR TITLE
[travis] add conda-4.3 to build matrix and fix test cases

### PR DESCRIPTION
**Pull request by jjhelmus**
purpose is to test upcoming changes in conda early.

### CODE REVIEW

**Comment by jjhelmus** [comment]
This is a submission of SciTools/conda-build-all#83 against the conda-forge fork of `conda-build-all`.  

@marscher Hopefully you do not mind that I submitted this for you.

**Comment by jakirkham** [comment]
Have you already enabled CIs? May need to (re)sync Travis CI to get this repo to show.

**Comment by jjhelmus** [comment]
> Have you already enabled CIs?

Not yet, probably should

**Comment by marscher** [comment]
thank you!

**Comment by jjhelmus** [comment]
> Have you already enabled CIs?

Is there some secret way of getting Travis CI to list the conda-forge repositories?  I get an error every time I try.

**Comment by jakirkham** [comment]
Yeah, their web app can't handle the number of repos we have. I have just now enabled it through the API. Should be good to go.

**Comment by jakirkham** [comment]
Toggling for CI.

**Comment by jjhelmus** [comment]
Thanks @jakirkham

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,12 @@ env:
           EXTRA_DEPS='conda-build=1.*'
         - PYTHON=3.5
           CONDA_BUILD_ALL_TEST_ANACONDA_CLOUD=1
+        - PYTHON=3.5
+          CONDA_ORIGIN=defaults
+
+matrix:
+  allow_failures:
+    - env: CONDA_ORIGIN=defaults
```
Comment (jakirkham) [inline .travis.yml:26]: Why do we have this `allow_failures`?

**Comment by marscher** [comment]
it should be removed. Was added for debugging purposes.

**Comment by jakirkham** [comment]
Thanks for the detail, @marscher. Will fix in a follow-up.

**Comment by jakirkham** [comment]
Thanks @marscher for putting this together, @jjhelmus for getting this onto conda-forge, and the many reviewers that have pushed this along.

**Pull request merged**